### PR TITLE
pthread -> thread_pool in Arb

### PR DESCRIPTION
**Pull request by albinahlback**
Should solve #1433

### CODE REVIEW

File: src/acb_mat/mul_threaded.c [hunk 1]
```diff
@@ -92,9 +90,30 @@ acb_mat_mul_threaded(acb_mat_t C, const acb_mat_t A, const acb_mat_t B, slong pr
         return;
     }
 
-    num_threads = flint_get_num_threads();
-    threads = flint_malloc(sizeof(pthread_t) * num_threads);
-    args = flint_malloc(sizeof(acb_mat_mul_arg_t) * num_threads);
+    num_workers = flint_request_threads(&handles, FLINT_MAX(ar, bc) - 1);
+    num_threads = num_workers + 1;
+
+    if (num_workers < 1)
```
Comment (fredrik-johansson) [inline src/acb_mat/mul_threaded.c:96]: I'm unsure whether this special case is really needed. We avoid a little bit of overhead but I don't think this is an issue for the situations where we actually call threaded matrix multiplication.

File: src/acb_poly/powsum_series_naive_threaded.c [hunk 2]
```diff
@@ -120,24 +122,24 @@ _acb_zeta_powsum_evaluator(void * arg_ptr)
...
+    _worker_arg * args;
+    slong i, num_threads, num_workers;
     int split_each_term;
 
-    num_threads = flint_get_num_threads();
+    split_each_term = (len > 1000);
 
-    threads = flint_malloc(sizeof(pthread_t) * num_threads);
-    args = flint_malloc(sizeof(powsum_arg_t) * num_threads);
+    num_workers = flint_request_threads(&handles, split_each_term ? len - 1 : n);
```
Comment (fredrik-johansson) [inline src/acb_poly/powsum_series_naive_threaded.c:139]: Is this supposed to be ``len - 1`` or ``len``?

Comment (albinahlback) [reply src/acb_poly/powsum_series_naive_threaded.c]: It is supposed to be `len - 1`, but it is also supposed to be `n - 1`. Thanks!

**Comment by fredrik-johansson** [comment]
I think the thread limit that you input to ``flint_request_threads`` is supposed to be the number of threads, not workers, even though it returns a number of workers. (This is certainly confusing, and it doesn't help that the function is undocumented.)

So it should actually be ``nz``, not ``nz - 1``, etc.

**Comment by albinahlback** [comment]
Allllrighhhtt, let's hope everything is good now haha

**Comment by fredrik-johansson** [comment]
Thanks, this looks perfect now (just hoping that CI passes).

**Pull request merged**